pkg/migrate/cstor/cstorvolumepolicy.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migrate
+
+import (
+	"strconv"
+	"strings"
+
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"gopkg.in/yaml.v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type config struct {
+	Name  string `yaml:"name"`
+	Value string `yaml:"value"`
+}
+
+type quantity struct {
+	Memory string `yaml:"memory,omitempty"`
+	CPU    string `yaml:"cpu,omitempty"`
+}
+
+// createCVPforConfig creates an equivalent CVP
+// for the cas config annotation set on old SC
+func (v *VolumeMigrator) createCVPforConfig(sc *storagev1.StorageClass) error {
+	_, err := v.OpenebsClientset.CstorV1().
+		CStorVolumePolicies(v.OpenebsNamespace).
+		Get(sc.Name, metav1.GetOptions{})
+	found := false
+	if err == nil {
+		found = true
+	}
+	if !k8serrors.IsNotFound(err) {
+		return err
+	}
+	scConfig := sc.Annotations["cas.openebs.io/config"]
+	scConfig = strings.TrimSpace(scConfig)
+	configs := []config{}
+	err = yaml.Unmarshal([]byte(scConfig), &configs)
+	if err != nil {
+		return err
+	}
+	cvp := &cstor.CStorVolumePolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sc.Name,
+			Namespace: v.OpenebsNamespace,
+		},
+	}
+	// populating the cvp with default spec,
+	// if any config exists on the sc annotation
+	// the below switch will override the default values
+	cvp.Spec = getDefaultCVPSpec()
```

Comment:
That is true when no policy is present for a storageclass. In case of migration, the user had policy configured with the help of annotation, now after migration, if some fields were not set it should have default value.

Reply:
I see the controller already has the code to default empty fields in the policy. I will revert the changes.